### DISCUSSION TITLE
Audio Compressor (FFMPEG)

### DISCUSSION THREAD

**Discussion by Fran355D** [iina/iina#3870] [Q&A] [answered]
Hi everyone, I hope you all are doing well.

First of all, I would like to apologise for my vague knowledge about all things related to commands and FFMPEG, and this is a new world for me.
Here I share these two images because that's what I'm looking for. I'm transitioning to IINA and would like the same settings as the dynamic audio compressor I had back in VLC. I tried that with the help of a wiki webpage, but the audio didn't work, and I think I'm doing something wrong here. Thanks in advance for your help. 

Kind regards from Uruguay

<img width="1010" alt="Captura de Pantalla 2022-07-12 a las 11 05 39 a  m" src="https://user-images.githubusercontent.com/102892785/178523470-b27b5793-649f-41e5-9e0a-f0b1ba993c36.png">
<img width="1440" alt="Captura de Pantalla 2022-07-12 a las 11 42 45 a  m" src="https://user-images.githubusercontent.com/102892785/178523494-6801b12a-23e5-4d74-8786-c00811e4efdd.png">

**Comment by low-batt** [comment, marked as answer]
Thanks for all the details in your question. Makes it much easier to understand the situation. 

First, be sure you are using the latest version of IINA, 1.3.0,  as some serious issue with filters have been fixed as of that release.

The question is about how to add a [FFmpeg](https://ffmpeg.org/) [acompressor](https://ffmpeg.org/ffmpeg-filters.html#acompressor) audio filter. This filter takes multiple options. The above screen shot shows multiple filters, each with one option set. We want to add only **one** FFmpeg audio filter and configure that filter with all the options needed.

In IINA select `Audio Filters...`, click `+` in the `Audio Filters` panel, click `Custom (FFmpeg)`. For `Filter name` specify `acompressor` for `Filter value` use:
```text
[threshold=20.6:ratio=20:attack=50:release=300:makeup=12:knee=1]
```

You can click `Save` and give the filter a name to reuse it next time you run IINA.

See if that works for you. Then adjust the filter settings as needed following the above cited FFmpeg documentation.

Feel free to enter a feature request issue requesting IINA provide an easy to use UI for this filter. The developers are overloaded, but it is always good to put in a request. It might catch the interest of a developer.

**Reply by Fran355D** [reply]
Thank you so much for such a detailed answer. I will try to enter a feature request issue as soon as possible.
